lib/util.js

```diff
@@ -72,6 +71,79 @@ const errorToString = Error.prototype.toString;
 var CIRCULAR_ERROR_MESSAGE;
 var Debug;
 
+// eslint-disable-next-line
+const escapeSequencesRegExp = /[\\'\x00-\x1f]/g;
+const keyStrRegExp = /^[a-zA-Z_][a-zA-Z_0-9]*$/;
+const textRegExp = /^[a-zA-Z_0-9 ,.?!+\-&"öäüÜÖÄß():;<>{}[\]=$]*$/;
+const colorRegExp = /\u001b\[\d\d?m/g;
+const numberRegExp = /^(0|[1-9][0-9]*)$/;
+
+// Escaped special characters. Use empty strings to fill up unused entries.
+const meta = [
+  '\\u0000', '\\u0001', '\\u0002', '\\u0003', '\\u0004',
+  '\\u0005', '\\u0006', '\\u0007', '\\b', '\\t',
+  '\\n', '\\u000b', '\\f', '\\r', '\\u000e',
+  '\\u000f', '\\u0010', '\\u0011', '\\u0012', '\\u0013',
+  '\\u0014', '\\u0015', '\\u0016', '\\u0017', '\\u0018',
+  '\\u0019', '\\u001a', '\\u001b', '\\u001c', '\\u001d',
+  '\\u001e', '\\u001f', '', '', '',
+  '', '', '', '', "\\'", '', '', '', '', '',
+  '', '', '', '', '', '', '', '', '', '',
+  '', '', '', '', '', '', '', '', '', '',
+  '', '', '', '', '', '', '', '', '', '',
+  '', '', '', '', '', '', '', '', '', '',
+  '', '', '', '', '', '', '', '\\\\'
+];
+
+// Escape control characters, single quotes and the backslash.
+// This is similar to JSON stringify escaping.
+function strEscape(str) {
+  // Some magic numbers that worked out fine while benchmarking
+  if (str.length < 5000 && textRegExp.test(str))
+    return `'${str}'`;
+  if (str.length > 50) {
+    return `'${str.replace(escapeSequencesRegExp, (str) => {
```

Comment:
Using a plain loop (with a `charCodeAt()` range check) instead *might* be faster than `replace()` + regexp?

Reply:
That part is explicitly added for strings with a length above 50 because of my benchmarks. It highly depended on the used input but in general the replace function wins for bigger string length.